src/big/big_decimal.cr

```diff
@@ -306,57 +306,93 @@ struct BigDecimal < Number
     self
   end
 
-  # Converts to `Int64`. Truncates anything on the right side of the decimal point.
-  def to_i64
+  # Converts to `BigInt`. Truncates anything on the right side of the decimal point.
+  def to_big_i
     if @value >= 0
-      (@value / TEN ** @scale).to_i64
+      (@value / TEN ** @scale)
     else
-      -(@value.abs / TEN ** @scale).to_i64
+      -(@value.abs / TEN ** @scale)
     end
   end
 
+  # Converts to `Int64`. Truncates anything on the right side of the decimal point.
+  def to_i64
+    to_big_i.to_i64
+  end
+
   # Converts to `Int32`. Truncates anything on the right side of the decimal point.
   def to_i32
-    to_i64.to_i32
+    to_big_i.to_i32
   end
 
   # Converts to `Int16`. Truncates anything on the right side of the decimal point.
   def to_i16
-    to_i64.to_i16
+    to_big_i.to_i16
   end
 
   # Converts to `Int8`. Truncates anything on the right side of the decimal point.
   def to_i8
-    to_i64.to_i8
+    to_big_i.to_i8
   end
 
   # Converts to `Int32`. Truncates anything on the right side of the decimal point.
   def to_i
     to_i32
   end
 
+  # Converts to `Int8`. Truncates anything on the right side of the decimal point.
+  def unsafe_to_i8
+    to_big_i.unsafe_to_i8
+  end
+
+  # Converts to `Int16`. Truncates anything on the right side of the decimal point.
+  def unsafe_to_i16
+    to_big_i.unsafe_to_i16
+  end
+
+  # Converts to `Int32`. Truncates anything on the right side of the decimal point.
+  def unsafe_to_i32
+    to_big_i.unsafe_to_i32
+  end
+
+  # Converts to `Int64`. Truncates anything on the right side of the decimal point.
+  def unsafe_to_i64
+    to_big_i.unsafe_to_i64
+  end
+
+  # Converts to `Int32`. Truncates anything on the right side of the decimal point.
+  def unsafe_to_i
+    unsafe_to_i32
+  end
+
+  # Converts to `BigInt`. Truncates anything on the right side of the decimal point,
+  # converting negative to positive.
+  def to_big_u
```

Comment:
Shouldn't this raise for negative values?

Reply:
Unsigned BitInt are not really a thing I will make that method private. This is not changing the current behavior which should also be reviewed because:

```crystal
-2.to_big_i.unsafe_to_u8 # => 2_u8
-2.unsafe_to_u8          # => 254_u8
```